### PR TITLE
remove apple-codesign crate

### PR DESCRIPTION
**Pull request by wolfv**

### CODE REVIEW

File: crates/rattler/src/install/link.rs
Comment (baszalmstra) [inline crates/rattler/src/install/link.rs:199]: I think we should return the error here instead of warning. Otherwise we leave the user with a non-functioning environment.

Comment (wolfv) [reply crates/rattler/src/install/link.rs:199]: Yeah, but then we can't try to execute this on Linux

Comment (baszalmstra) [reply crates/rattler/src/install/link.rs:199]: Yeah thats fine right? Because the environment is not valid anyway? Otherwise I would suggest we add an option to disable signing or ignore signing errors. But by default I think this is an error that should bubble up.

Comment (wolfv) [reply crates/rattler/src/install/link.rs:199]: Or we should not try to sign packages when we don't find the binary?

Comment (baszalmstra) [reply crates/rattler/src/install/link.rs:199]: Maybe I dont get it, but that would still silently result in an environment that doesnt work on regular arm64 macs right?

Comment (wolfv) [reply crates/rattler/src/install/link.rs:199]: Hypothetically, if you were to cross-compile from Linux to osx-arm64 you don't need to have signed binaries (shared libs also need signing). And you wouldn't try to execute the code.

Comment (baszalmstra) [reply crates/rattler/src/install/link.rs:199]: But in that (edge) case I would prefer to set an option to not sign the binaries at all.

Comment (baszalmstra) [reply crates/rattler/src/install/link.rs:199]: If you use this on an arm64 mac and for some reason it fails the user should be notified that it failed.

**Review by baszalmstra — APPROVED** [review]

**Comment by baszalmstra** [comment]
@pavelzw So in the end we still ended up removing the apple-codesign binary. 

- The build was utterly failing on the crate.
- The maintainer doesn't seem very open to fixing our issue.
- Cross-creating environments for Mac with ad-hoc certificates will still not work on another mac (defeating its purpose completely)

We ended up using the codesign binary on Mac.

The upside is that maybe now the other targets you were trying will work?

**Comment by pavelzw** [comment]
So ppc64le still has hope 🤪 I'll get right to it once this is merged 😎

**Pull request merged**
